pkg/api/types.go

```diff
@@ -673,6 +673,9 @@ type GlusterfsVolumeSource struct {
 	// Required: EndpointsName is the endpoint name that details Glusterfs topology
 	EndpointsName string `json:"endpoints"`
 
+	// Optional: Servers is the gluster server list
+	EndpointsNameSpace *string `json:"endpointsns,omitempty"`
```

Comment:
What if we keep the current behavior (no endpoint namespace) for VolumeSource in POD, and enforce with admission controller, but allow the PV.GVS.endpoint-name and PV.GVS.endpoint-namespace since PV is an admin object.  @eparis, didn't you mention that a similar pattern was recently used by another kube component?